clang/tools/driver/CacheLauncherMode.cpp

```diff
@@ -41,11 +42,26 @@ static bool shouldCacheInvocation(ArrayRef<const char *> Args,
       llvm::remove_if(CheckArgs, [](StringRef Arg) { return Arg == "-###"; }),
       CheckArgs.end());
   CreateInvocationOptions Opts;
-  Opts.Diags = Diags;
```

Comment:
What if just create a new `DiagnosticsEngine` with `IgnoreDiagConsumer`? Then `Diags.hasErrorOccurred()` should not cache?

Reply:
@akyrtzi you set this code up to pass in the diagnostics engine; is it okay to just ignore errors here instead?

Reply:
I mean used a different diagnostics engine for `createInvocation` and just for that purpose. The warning from this function is still emitted to the engine passed in.

Reply:
I think it would be fine to ignore errors; if we find it a reason later on we can go back to adding explicit `IgnoreMissingCompile`.

Reply:
Done in latest update: we now ignore the diagnostics from parsing options entirely in `shouldCacheInvocation`. We will still emit diagnostics for `warn_clang_cache_disabled_caching` for specific cases we disable caching.